firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/CrashlyticsController.java

```diff
@@ -860,7 +860,7 @@ private void doCloseSessions(int maxCustomExceptionEvents, boolean includeCurren
 
     closeOpenSessions(sessionBeginFiles, offset, maxCustomExceptionEvents);
 
-    reportingCoordinator.finalizeSessions();
+    reportingCoordinator.finalizeSessions(new Date().getTime() / 1000);
```

Comment:
This didn't break any CrashlyticsController tests. Do we not mock reportingCoordinator in those tests to check it was called in unit?

... one pattern I often see (and implement) is a TimeProvider interface that would do new Date().getTime() / 1000, what's nice about that pattern is then you could pass it in and mock it to make the tests not break (if they had)

Reply:
I considered that pattern, and in fact we actually have a `SystemCurrentTimeProvider`, but I wasn't sure whether to pass it in here or into the constructor.

Reply:
Cool, there's probably a wider discussion about that which I'm happy to punt on, but we can chat about when you're interested.

Given that this change didn't break any tests, do you need to add any tests that validate this method?

Reply:
I think we can do a separate pass and add tests for validating all of the `SessionReportingCoordinator` calls in `CrashlyticsController`.